### PR TITLE
gatekeeper: take care of fragmented packets

### LINKED ISSUE #307: Gatekeeper must be on the lookout for fragmented packets
Most of the code of Gatekeeper only looks up to the network headers of packets, but some of the code looks above network headers, so this portion of the code must deal with the possibility that the packet is fragmented. For example, packets carrying policy decisions from Grantor servers to Gatekeeper servers may be fragmented in the network.

In the case of a fragmented GGU packet, Gatekeeper should discard the packet and log the event. Any other case in which dealing with fragmented packets is needed, has to be evaluated case by case.

One basic search to find places to deal with fragmented packets is grepping for `ipv4_skip_exthdr()` and `ipv6_skip_exthdr()`. Notice that this search is not going to be complete. To see something that wouldn't be listed, see `process_single_packet()` in file `ggu/main.c`, it doesn't call these functions because it assumes that the packets are **not** fragmented. The basic search should be complemented with a general analysis of the functional blocks.

### PR DESCRIPTION
**Pull request by mengxiang0811**
This patch closes #307

### CODE REVIEW

File: cps/main.c [hunk 1]
```diff
@@ -918,6 +918,13 @@ match_bgp4(struct rte_mbuf *pkt, struct gatekeeper_if *iface)
 	if (ip4hdr->next_proto_id != IPPROTO_TCP)
 		return -ENOENT;
 
+	if (rte_ipv4_frag_pkt_is_fragmented(ip4hdr)) {
+		CPS_LOG(WARNING,
+			"Received IPv4 fragmented packets destined to the Gatekeeper server at %s\n",
```
Comment (AltraMayor) [inline cps/main.c]: A CPS block may be present in a Grantor server, so the message is misleading.

Please review this and all the other log messages.

File: lls/main.c [hunk 2]
```diff
@@ -503,6 +524,13 @@ match_ping6(struct rte_mbuf *pkt, struct gatekeeper_if *iface)
 			sizeof(iface->ll_ip6_mc_addr)) != 0))
 		return -ENOENT;
 
+	if (rte_ipv6_frag_pkt_is_fragmented(ip6hdr)) {
```
[the remark this answers is not in the document]
Comment (AltraMayor) [reply lls/main.c]: If you choose to add a parameter to `ipv6_skip_exthdr()` to find out if the packet is fragmented, do so as a new patch to this pull request. That is, this pull request would have two patches. The new patch should come before the current patch.

[the remark this answers is not in the document]
Comment (mengxiang0811) [reply lls/main.c]: We can use ``static struct ipv6_extension_fragment* rte_ipv6_frag_get_ipv6_fragment_header (struct ipv6_hdr * hdr)`` to check whether it's a fragmented packet.

**Comment by mengxiang0811** [comment]
This rebase is ready for another review.

**Pull request merged**